File: MobileServer/src/main/java/com/example/MobileServer/domain/Expense.java
package com.example.MobileServer.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "Expense")
public class Expense {

    @Id
    @Column(name = "_id")
    private Integer id;
    private String title;
    private String description;
    private Integer amount;
    private String category;
    private Date date;
    private String payment_method;

}

File: SmartFinNative4/app/src/main/java/com/example/smartfinnative4/model/Expense.kt
package com.example.smartfinnative4.model

import android.util.Log
import java.io.Serializable
import java.time.LocalDate

data class Expense(var title: String, var description: String, var amount: Int, var category: String,
                   var date: LocalDate, var payment_method: String, var id: Int): Serializable
{
    companion object {
        var currentId = 0
    }

    constructor(
        title: String, description: String, amount: Int, category: String, date: LocalDate, payment_method: String
    ) : this(
        title, description, amount, category, date, payment_method, currentId++
    ) {
        Log.i("Model Expense class: ", "CurrentId is $currentId")
    }


}
